File: NVR_Client_App/90116/ActiveX-Sample/Sample Program/PS-LOOKUP/Visual CSharp/301_PSLookup/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace SimpleSample_PSLookup
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        //---------------------------------------------------------------------
        // Define variables
        //---------------------------------------------------------------------
        private Form2 frmForm2;
        //SearchStatus 0:Off 1:On
        enum SearchStatus {
            SearchOff = 0,
            SearchOn = 1
        };
        SearchStatus gStatus;

        //****************************************************************
        //* Function Name        : Load
        //* Overview             : Load PSAPI and Initialize
        //****************************************************************
        private void Form1_Load(object sender, EventArgs e)
        {
            //Show the log window
            frmForm2 = new Form2();
            frmForm2.Location = new Point(this.Size.Width, 0);
            frmForm2.Show();

            //----------------------------------------
            //Set properties (ActiveX)
            //----------------------------------------
            // Set non visible
            AxpsLookupCtrl.Visible = false;
            AxpsLookupCtrl.Height = 10;
            AxpsLookupCtrl.Width = 10;

            //Set properties for event
            AxpsLookupCtrl.OnDevLookupEnable = 0;
            Logging("[Property] OnDevLookupEnable:=0");
            AxpsLookupCtrl.OnErrorEnable = 1;

            //----------------------------------------
            //Set parameters (Other controls)
            //----------------------------------------
            listBox1.Items.Clear();
  
        }

        //****************************************************************
        //* Function Name        : FormClosed
        //* Overview             : Destroy main window
        //****************************************************************
        private void Form1_FormClosed(object sender, FormClosedEventArgs e)
        {
            //----------------------------------------
            //Set properties (ActiveX)
            //----------------------------------------
            AxpsLookupCtrl.OnDevLookupEnable = 0;
            Logging("[Property] OnDevLookupEnable:=0");
            AxpsLookupCtrl.OnErrorEnable = 0;
            
            frmForm2.Close();
        }

        //****************************************************************
        //* Function Name        : Search_Click
        //* Overview             : Switching the camera searches
        //****************************************************************
        private void b_Search_Click(object sender, EventArgs e)
        {
            if (gStatus == SearchStatus.SearchOff) {
                AxpsLookupCtrl.OnDevLookupEnable = 1;
                Logging("[Property] OnDevLookupEnable:=1");
                gStatus = SearchStatus.SearchOn;
                b_Search.Text = "&End";
            }
            else {
                this.AxpsLookupCtrl.OnDevLookupEnable = 0;
                Logging("[Property] OnDevLookupEnable:=0");
                gStatus = SearchStatus.SearchOff;
                b_Search.Text = "&Start";
            }
        }

        //****************************************************************
        //* Function Name        : OnDevLookup
        //* Overview             : OnDevLookup listener event
        //****************************************************************
        private void AxpsLookupCtrl_OnDevLookup(object sender, AxpslookupLib._IpslookupctrlEvents_OnDevLookupEvent e)
        {
            String Logs;
            //Output Logs
            Logs = "macAddr : " + e.macAddr + "\r\n" +
                    "ipAddr : " + e.ipAddr + "\r\n" +
                    "ipv6Addr : " + e.ipv6Addr + "\r\n" +
                    "portNo : " + e.portNo + "\r\n" +
                    "camName : " + e.camName + "\r\n" +
                    "modelName : " + e.modelName;
            Logging("[OnDevLookup] " + Logs);

            Logs = Logs.Replace("\r\n", " ");
            listBox1.Items.Add(Logs);
        }

        //****************************************************************
        //* Function Name        : OnError
        //* Overview             : OnError listener event
        //****************************************************************
        private void AxpsLookupCtrl_OnError(object sender, AxpslookupLib._IpslookupctrlEvents_OnErrorEvent e)
        {
            //Output Logs
            Logging("[OnError] errorCode[" + e.errorCode.ToString() + "] description[" + e.description + "]");
        }

        //****************************************************************
        //* Function Name        : Logging
        //* Overview             : Output Logs
        //****************************************************************
        private void Logging(string str)
        {
            if (frmForm2.textLog.Text.CompareTo(string.Empty) != 0) {
                frmForm2.textLog.Text = frmForm2.textLog.Text + "\r\n";
            }
            frmForm2.textLog.Text = frmForm2.textLog.Text + str;
        }

    }
}

File: NVR_Client_App/90116/ActiveX-Sample/Sample Program/PS-LOOKUP/Visual Basic 2012/301_PSLookup/Form1.vb
Public Class Form1
    '****************************************************************
    '* Define
    '****************************************************************
    'SearchStatus 0:Off 1:On
    Enum SearchStatus
        SearchOff = 0
        SearchOn = 1
    End Enum
    Public gStatus As SearchStatus

    '****************************************************************
    '* Function Name        : Load
    '* Overview             : Load PSAPI & Initialize
    '****************************************************************
    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        'Show the log window
        Form2.Location = New Point(Me.Size.Width, 0)
        Form2.Show()

        'Set initial status
        gStatus = SearchStatus.SearchOff

        '----------------------------------------
        'Set properties (ActiveX)
        '----------------------------------------
        With AxpsLookupCtrl
            ' Set non visible
            .Visible = False
            .Height = 10
            .Width = 10

            'Set properties for event
            .OnDevLookupEnable = 0
            Logging("[Property] OnDevLookupEnable:=0")
            .OnErrorEnable = 1
        End With

        '----------------------------------------
        'Set parameters (Other controls)
        '----------------------------------------
        ListBox1.Items.Clear()
        ListBox1.HorizontalScrollbar = True


    End Sub

    '****************************************************************
    '* Function Name        : FormClosed
    '* Overview             : Destroy main window
    '****************************************************************
    Private Sub Form1_FormClosed(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles MyBase.FormClosed
        '----------------------------------------
        'Set properties (ActiveX)
        '----------------------------------------
        With AxpsLookupCtrl
            'Set properties for event
            .OnDevLookupEnable = 0
            .OnErrorEnable = 0
        End With
        Form2.Close()
    End Sub

    '****************************************************************
    '* Function Name        : Search_Click
    '* Overview             : Switching the camera searches
    '****************************************************************
    Private Sub b_Search_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles b_Search.Click
        If gStatus = SearchStatus.SearchOff Then
            AxpsLookupCtrl.OnDevLookupEnable = 1
            Logging("[Property] OnDevLookupEnable:=1")
            gStatus = SearchStatus.SearchOn
            b_Search.Text = "&End"
        Else
            AxpsLookupCtrl.OnDevLookupEnable = 0
            Logging("[Property] OnDevLookupEnable:=0")
            gStatus = SearchStatus.SearchOff
            b_Search.Text = "&Start"
        End If
    End Sub

    '****************************************************************
    '* Function Name        : OnDevLookup
    '* Overview             : OnDevLookup listener event
    '****************************************************************
    Private Sub AxpsLookupCtrl_OnDevLookup(ByVal sender As System.Object, ByVal e As AxpslookupLib._IpslookupctrlEvents_OnDevLookupEvent) Handles AxpsLookupCtrl.OnDevLookup
        Dim Logs As String
        'Output Logs
        Logs = "macAddr : " & e.macAddr & vbCrLf & _
                "ipAddr : " & e.ipAddr & vbCrLf & _
                "ipv6Addr : " & e.ipv6Addr & vbCrLf & _
                "portNo : " & e.portNo & vbCrLf & _
                "camName : " & e.camName & vbCrLf & _
                "modelName : " & e.modelName
        Logging("[OnDevLookup] " & Logs)
        Logs = Logs.Replace(vbCrLf, " ")
        ListBox1.Items.Add(Logs)
    End Sub

    '****************************************************************
    '* Function Name        : OnError
    '* Overview             : OnError listener event
    '****************************************************************
    Private Sub AxpsLookupCtrl_OnError(ByVal sender As System.Object, ByVal e As AxpslookupLib._IpslookupctrlEvents_OnErrorEvent) Handles AxpsLookupCtrl.OnError
        'Output Logs
        Logging("[OnError] errorCode[" & e.errorCode.ToString() & "] description[" & e.description & "]")
    End Sub

    '****************************************************************
    '* Function Name        : Logging
    '* Overview             : Output Logs
    '****************************************************************
    Private Sub Logging(ByVal str As String)
        If Form2.txt_Log.Text <> String.Empty Then
            Form2.txt_Log.Text = Form2.txt_Log.Text & vbCrLf
        End If
        Form2.txt_Log.Text = Form2.txt_Log.Text & str
    End Sub

End Class
